### PR TITLE
Upgrade: some (dev) deps

### PR DESCRIPTION
**Pull request by aladdin-add**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [x] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

#### What changes did you make? (Give an overview)
as the title.

#### Is there anything you'd like reviewers to focus on?

1. after upgrading eslint-plugin-jsdoc, it reports:
  "Invalid JSDoc tag name "template"  jsdoc/check-tag-names"
  it was fixed by adding `setting.jsdoc.mode: "typescript"`
  (refs:https://github.com/gajus/eslint-plugin-jsdoc#check-tag-names)

2. the following deps were not upgraded:
* levn (was upgraded in eslint/eslint#13140)
* ignore (it has a few breaking changes)

### CODE REVIEW

File: packages/eslint-config-eslint/package.json [hunk 1]
```diff
@@ -20,8 +20,8 @@
   "homepage": "https://eslint.org",
   "bugs": "https://github.com/eslint/eslint/issues/",
   "peerDependencies": {
-    "eslint-plugin-jsdoc": "^15.9.5",
-    "eslint-plugin-node": "^9.0.0"
```
Comment (kaicataldo) [inline packages/eslint-config-eslint/package.json:24]: Ah, thanks for catching this. I missed this in my PR.

**Comment by aladdin-add** [comment]
I've rebased master. @kaicataldo

File: lib/shared/relative-module-resolver.js [hunk 2]
```diff
@@ -11,8 +11,7 @@ const Module = require("module");
  * `Module.createRequire` is added in v12.2.0. It supports URL as well.
  * We only support the case where the argument is a filepath, not a URL.
  */
-// eslint-disable-next-line node/no-unsupported-features/node-builtins, node/no-deprecated-api
-const createRequire = Module.createRequire || Module.createRequireFromPath;
+const createRequire = Module.createRequire || Module.createRequireFromPath; // eslint-disable-line node/no-unsupported-features/node-builtins, node/no-deprecated-api
```
Comment (kaicataldo) [inline lib/shared/relative-module-resolver.js:14]: Can we keep these as two separate lines? Easier to read :)

File: .eslintrc.js [hunk 3]
```diff
@@ -53,6 +53,11 @@ module.exports = {
     parserOptions: {
         ecmaVersion: 2020
     },
+    settings: {
+        jsdoc: {
+            mode: "typescript"
```
Comment (kaicataldo) [inline .eslintrc.js:58]: Why does this need to be changed from the default of `jsdoc`?

File: lib/shared/relative-module-resolver.js (hunk 2 above)
Comment (aladdin-add) [reply lib/shared/relative-module-resolver.js:14]: as we are using tag `template`, see https://github.com/gajus/eslint-plugin-jsdoc#check-tag-names

File: .eslintrc.js (hunk 3 above)
Comment (aladdin-add) [reply .eslintrc.js:58]: it reports: "Invalid JSDoc tag name "template" jsdoc/check-tag-names"
as we are using tag `template`, see https://github.com/gajus/eslint-plugin-jsdoc#check-tag-names

Comment (aladdin-add) [reply .eslintrc.js:63]: added a comment. 😄

**Comment by nzakas** [comment]
@kaicataldo has your feedback been addressed?

**Pull request merged**